src/mon/NVMeofGwSerialize.h

```diff
@@ -254,12 +255,30 @@ inline void decode(ana_state_t& st, ceph::buffer::list::const_iterator &bl) {
   DECODE_FINISH(bl);
 }
 
-inline void encode(const GwSubsystems& subsystems,  ceph::bufferlist &bl) {
-  ENCODE_START(1, 1, bl);
+inline void encode(
+  const GwSubsystems& subsystems,  ceph::bufferlist &bl, uint64_t features) {
+  uint8_t version = 1;
+  if (HAVE_FEATURE(features, SERVER_SQUID)) {
+    version = 2;
+  }
+  ENCODE_START(version, 1, bl);
```

Comment:
This should be version, version -- version 1 decoders cannot decode version 2.  Check every other usage.

Reply:
will do